Subject: [patch] hostfs uml add some other pagecache methods.patch
From: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Patch-mainline: 2.6.11-rc1

This is a follow-up to my previous "hostfs - uml: set .sendfile to
generic_file_sendfile" patch. I was asking whether other methods should have
been added, and comparing with ext3 I found some more ones.

However, I have not specific clues about them: I know they use the pagecache,
which relies on *page methods, which are defined by hostfs. So I think it
could work.

I have a doubt, whether hostfs needs the commented out method below:

static struct address_space_operations hostfs_aops = {
	.writepage 	= hostfs_writepage,
	.readpage	= hostfs_readpage,
/* 	.set_page_dirty = __set_page_dirty_nobuffers, */
	.prepare_write	= hostfs_prepare_write,
	.commit_write	= hostfs_commit_write
};

Hostfs does not have a underlying device (and I have some rough idea that
buffers cache block devices data), so I wonder if that is needed or not.

Signed-off-by: Paolo 'Blaisorblade' Giarrusso <blaisorblade_spam@yahoo.it>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Gerd Knorr <kraxel@suse.de>
---
 fs/hostfs/hostfs_kern.c |    5 +++++
 1 files changed, 5 insertions(+)

diff -puN fs/hostfs/hostfs_kern.c~hostfs-uml-add-some-other-pagecache-methods fs/hostfs/hostfs_kern.c
--- 25/fs/hostfs/hostfs_kern.c~hostfs-uml-add-some-other-pagecache-methods	2004-11-18 23:42:01.024121960 -0800
+++ 25-akpm/fs/hostfs/hostfs_kern.c	2004-11-18 23:42:01.028121352 -0800
@@ -394,6 +394,10 @@ static struct file_operations hostfs_fil
 	.llseek		= generic_file_llseek,
 	.read		= generic_file_read,
 	.sendfile	= generic_file_sendfile,
+	.aio_read	= generic_file_aio_read,
+	.aio_write	= generic_file_aio_write,
+	.readv		= generic_file_readv,
+	.writev		= generic_file_writev,
 	.write		= generic_file_write,
 	.mmap		= generic_file_mmap,
 	.open		= hostfs_file_open,
@@ -402,6 +406,7 @@ static struct file_operations hostfs_fil
 };
 
 static struct file_operations hostfs_dir_fops = {
+	.llseek		= generic_file_llseek,
 	.readdir	= hostfs_readdir,
 	.read		= generic_read_dir,
 };
_
